common/scala/src/main/scala/whisk/spi/SpiProvider.scala

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.spi
+
+import java.util.ServiceLoader
+
+import akka.actor.ActorSystem
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import scaldi.Binding
+import scaldi.DynamicModule
+import scaldi.Identifier
+import scaldi.ImmutableWrapper
+import scaldi.Injectable
+import scaldi.Injector
+import scaldi.MutableInjector
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ListBuffer
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
+
+/**
+ * An Spi (Service Provider Interface) is an extension point.
+ * At runtime, multiple Spi implementations may exist in the classpath, but only a single implementation
+ * will be available for consumers to use. The specific implementation available at runtime is governed by the
+ * configKey specified in the SpiProvider
+ */
```

Comment:
Do you intend for the Spi trait to be more than a marker interface in the future?